python/common/org/python/types/Str.java

```diff
@@ -1296,14 +1296,78 @@ private long toPositiveIndex(long index) {
     }
 
     @org.python.Method(
-            __doc__ = "S.splitlines([keepends]) -> list of strings\n" +
-                    "\n" +
-                    "Return a list of the lines in S, breaking at line boundaries.\n" +
-                    "Line breaks are not included in the resulting list unless keepends\n" +
-                    "is given and true.\n"
-    )
-    public org.python.Object splitlines() {
-        throw new org.python.exceptions.NotImplementedError("splitlines() has not been implemented.");
+            __doc__ = "S.splitlines([keepends]) -> list of strings\n\nReturn a list of the lines in the string, breaking at line boundaries. Line breaks are not included in the resulting list unless keepends is given and true.",
+            default_args = "keepends"
+    )
+    public org.python.Object splitlines(org.python.Object keepends) {
+        boolean parseKeepEnd;
+        if (keepends == null) {
+            parseKeepEnd = false;
+        } else if (keepends instanceof org.python.types.Bool) {
+            parseKeepEnd = ((org.python.types.Bool) keepends).value;
+        } else if (keepends instanceof org.python.types.Int) {
+            parseKeepEnd = ((org.python.types.Int) keepends).value != 0;
+        } else if (keepends instanceof org.python.types.Float) {
```

Comment:
It seems that Int-like objects are accepted: (but not Bool-like ones)

```
class IntLike:
    def __int__(self):
        return 5

x = "abc\ndef"
y = IntLike()
x.splitlines(y)
x.splitlines(0)
```

It might be simpler to call `__int__()` on `keepends`, then test for `!= 0`. I think that will take care of the Bool, Int, Float, and Complex cases.